README.md

```diff
@@ -14,54 +16,59 @@ Run `rails g async_request`
 
 ### Usage
 
-Simply call `async_request(your_worker, *args)` like this:
+This gem provides a  that will call an `execute` a method of any class you give and will store the response in the database. In order to do this, we provide a model named `AsyncRequest::Job` that will handle the queue of the worker with the params.
+
+* IMPORTANT: The response of the worker will store the response as a string and the status code as an integer.
 
 ``` ruby
-class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
-  protect_from_forgery with: :exception
-
-  def my_compute_heavy_endpoint
-    id = execute_async(MyComputeHeavyWorker, { some: 'args' }, 'another arg')
-    render json: { id: id, url: async_request.job_url(id) }, status: 202
+class SomeController < ApplicationController
+  def my_compute_heavy_endpoint_option_1
+    _job, token = AsyncRequest::Job.execute_async(MyComputeHeavyWorker, { some: 'args' }, 'another arg')
+    render json: { token: token, url: async_request.job_url }, status: :accepted
+  end
+
+  def my_compute_heavy_endpoint_option_2
+    _job, token = AsyncRequest::Job.execute_async(MyComputeHeavyWorker, { some: 'args' }, 'another arg')
+    render json: { token: token }, status: :accepted, location: async_request.job_url
+  end
+end
+
+class MyComputeHeavyWorker
+  def execute(options, string)
+    # Perform heavy task
+    [:ok, { message: 'success' }.to_json] # returning 200 is valid too
```

Comment:
All the responses would have to respond to `to_json` since we are doing `JSON.parse` when returning them. Why don't we execute the `.to_json` in the gem ? If not, the user would have to do it manually every time.